Redesigned Marketplace

JorgeVidal29 opened the pull request:
**Description**
Redesigned Marketplace

Fixes #(issue)

**Type of change**
Check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Fix or feature that would cause existing functionality to not work as expected
- [ ] This change requires a update in the design document

**Checklist:**
Go down the list and check them off.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the design document (if applicable)
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

Zexdur committed:
Updated the marketplace view

The marketplace has a new look.

Zexdur committed:
Update Marketplace.js

Zexdur committed:
Update MarketPlace

Zexdur committed:
Update to MarketPlace

Need to talk to Adam to see if we can remove the html body color.

SecondThread commented on fortuna/src/Main.css:
```diff
@@ -0,0 +1,106 @@
+.styleForRow {
```
Why are there a bunch of additions on this file if it already existed in master? It makes it hard to see what to focus on when reviewing code.

JorgeVidal29 replied:
Hey for the changes, the file hadn't been put into master yet when I was working. I can delete mine. The login btn is probably a adam question.

blaiadam replied:
Jorge, we should collaborate on the CSS stuff sometime soon.

SecondThread commented on fortuna/src/Main.css:
```diff
@@ -0,0 +1,106 @@
+
+
+/* Button Styling */
+.loginbtn {
```
If this is in a global CSS file, then these styles should be styles of all buttons. That is, we should have like one main button (like maybe blue or something) and one secondary button (for 'cancel' or whatever). But we definitely shouldn't have stuff specific to login in a global CSS file.

SecondThread commented on fortuna/src/Main.css:
```diff
@@ -0,0 +1,106 @@
+    width: 25%;
+}
+
+.loginbtn:hover {
```
Same as above

SecondThread commented on fortuna/src/Main.css:
```diff
@@ -0,0 +1,106 @@
+	border: 2px solid #04CCFF;
+}
+
+.signupbtn {
```
Same as above; this should be reusable so we don't need new CSS for every button in this entire app

SecondThread commented on fortuna/src/Main.css:
```diff
@@ -0,0 +1,106 @@
+    background: none;
+}
+
+.signupbtn:hover {
```
same as above, but you get the point, so I'll stop

SecondThread commented on fortuna/src/Main.css:
```diff
@@ -0,0 +1,106 @@
+
+
+/* Popup Styling */
+.loginpopup {
```
All these popups are the same. There is no reason we should have 3 different classes that all mean the same thing and are all used for the same thing.

SecondThread commented on fortuna/src/Main.css:
```diff
@@ -0,0 +1,106 @@
+}
+
+.list-group-item list-group-item-action{
+    color:"black";
```
Put a space after your colon please

SecondThread commented on fortuna/src/marketplace/AccessoriesView.js:
```diff
@@ -0,0 +1,13 @@
+
+class AccessoriesView extends React.Component<{||}> {
+    render(){
+		return (
```
Fix the indentation here please

SecondThread commented on fortuna/src/marketplace/CannonsView.js:
```diff
@@ -0,0 +1,13 @@
+
+class CannonsView extends React.Component<{||}> {
+    render(){
+		return (
```
same as above, and same for the rest of the files

SecondThread commented on fortuna/src/marketplace/AccessoriesView.js:
```diff
@@ -0,0 +1,13 @@
+		return (
+            <h2>This is the Accessories View</h2>
+            );
+	  }
```
indentation with these two lines is off too

SecondThread commented on fortuna/src/marketplace/CasusBlocksView.js:
```diff
@@ -0,0 +1,13 @@
+import * as React from 'react';
+import 'bootstrap/dist/css/bootstrap.min.css';
+
+class CasusBlockView extends React.Component<{||}> {
```
Just curious: what is the Casus Block View going to show? The Casus stuff is going to be in the CasusContainer component.

JorgeVidal29 replied:
This is going to be the view for selling stuff that does with Casus on the marketplace that's why its under the marketplace folder

SecondThread commented on fortuna/src/marketplace/Marketplace.js:
```diff
@@ -3,9 +3,58 @@
+
+type Props = {||}; 
+type State = {|
+	marketplaceView: React.Element<any>,
```
This is very bad for many reasons.

First: It isn't a good idea to use any, or object unless you absolutely have to. It isn't 'any' type; we know what type it is.

Second (and more importantly): This destroys how react works. Storing the component that we want to render in our state makes it so that when react components get re-rendered the internal components don't actually get re-rendered. The state of the component should be the _information_ about what gets rendered (such as which tab is selected). It shouldn't be the _thing itself_ that is rendered.

Here are some problems that creating components and storing them in your state instead of creating them in the render method causes:
- render() won't act as expected when components are re-rendered. 
- If the props/state of this component update, we will likely need to update the props of a child state. We can't do this if that is stored in our state, without really complex logic in OnComponentDidUpdate.
- We might not know what props we need to pass to this child component until we are in the render method. We are creating the component that we will end up rendering sometime way later and storing it in our state. What do we pass to that component when we create it?

Instead of storing the component, you could store the selectedTankPartType. Then, when the component is actually rendering, you can create the SomethingView that you need to create.

SecondThread commented on fortuna/src/marketplace/Marketplace.js:
```diff
@@ -3,9 +3,58 @@
+			marketplaceView: <h2>Select a type on the left</h2>
+		}
+	}
+	showMarketView(selectedTankPartType: string) {
```
If there are a fixed number of strings that this is allowed to be, this needs to be a type (i. e. a type that isn't string). Why is this bad? What if someone called this method and passed in 'Cannons' for example? Flow would say that is just fine, but that clearly isn't acceptable input. The fact that it isn't should be indicated by its method signature.

SecondThread replied:
Also you need to label the return type of this method. (looks like void)

SecondThread commented on fortuna/src/marketplace/Marketplace.js:
```diff
@@ -21,31 +70,25 @@ class Marketplace extends React.Component<{||}> {
 	  		</div>
 		);
-  	}
+	  }
```
I can't tell if this line used to have bad indentation or now has bad indentation, but it doesn't line up with the diff, so make sure it is right now.

Derplime commented:
While also doing the changes proposed by @SecondThread, you'll need to update your branch to the new file hierarchy, since that was recently merged in.

blaiadam commented on fortuna/src/marketplace/ChassisView.js:
```diff
@@ -0,0 +1,13 @@
+//@flow strict
+import * as React from 'react';
+import 'bootstrap/dist/css/bootstrap.min.css';
```
So with the way that our web page is functioning, as long as this CSS import is included in Login.js or App.js, it will be imported on all of our js files. This renders the import of the bootstrap CSS file here unnecessary. You should be able to safely remove this from all of the files in this PR.

Zexdur committed:
Updated Marketplace

Marketplace was given different views. This was updated to be up to standard with review

SecondThread reviewed (changes requested):
You gotta pull from master bud. We don't want 108 files changed from files you haven't touched.

![image](https://user-images.githubusercontent.com/18317476/73621143-eee3b200-4602-11ea-875f-18bc6ad8dc4f.png)

run
git pull origin master

then push again
